### ISSUE TITLE
R101-FPN results unreproducible

### ISSUE THREAD

**Issue by jason718** [roytseng-tw/Detectron.pytorch#77]
I tried to train R101-FPN-1x / R101-FPN-2x using the default setting. Unfortunately, both of them cannot reproduce the results in Detectron.caffe2. (~2%-3% lower)

Any idea what happened? Why R50-FPN-1x is good but these are not?

**Comment by jason718** [comment]
This is R-101-FPN-1x results

![image](https://user-images.githubusercontent.com/3235510/41117456-ef45c592-6a41-11e8-96a9-19a5e87ef6d1.png)

**Comment by jason718** [comment]
R-101-FPN-2x results:

![coco_repro_101fpn_2x](https://user-images.githubusercontent.com/3235510/41119687-42f51b60-6a48-11e8-8479-ae183a95f1a1.png)

**Comment by roytseng-tw** [comment]
Hi, thank you for reporting this. 
I am not sure about the reason right now, because I haven't personally trained a R-101 baseline.
Can you provide me with your exact training command and environment ?
I'll check this later.

**Comment by jason718** [comment]
I am using 8 V100 GPUs,
python3 tools/train_net_step.py \
    --dataset coco2017 \
    --cfg configs/baselines/e2e_mask_rcnn_R-101-FPN_2x.yaml \
    --use_tfboard \
    --bs 16 --nw 16

**Comment by li-js** [comment]
I also experienced the similar issue, e2e_mask_rcnn_R-101-FPN_2x.yaml gives results about 2 points lower than official Detection as in #64 

The same thing goes to the e2e_mask_rcnn_X-152-32x8d-FPN-IN5k_1.44x.yaml, which gives 37.68 Segm AP (no test aug), compared to official Detectron's 39.7, it is also about 2 points lower. 

Trained with 4 GPU:
python3 tools/train_net_step.py 
--dataset coco2017 
--cfg configs/baselines/e2e_mask_rcnn_X-152-32x8d-FPN-IN5k_1.44x.yaml
--bs 4 --iter_size 2

Results:
![image](https://user-images.githubusercontent.com/9071670/41271771-f3d79cbe-6e44-11e8-97a1-2b849e507bd5.png)


I guess there is something going on here for deep models for this bias .

**Comment by slowbull** [comment]
hi, does anyone solve this issue?

**Comment by jason718** [comment]
@slowbull the point is I don't even know how to solve it. I actually don't understand why there is difference between different resnet models.

**Comment by roytseng-tw** [comment]
Hello guys, after fixed the resnet pretrained weights loading problem on latest commit, I managed to get 39.2 on e2e_faster_rcnn-R-101-FPN_1x. It's way higher than 38.5 in Detectron, and even better than the 39.0 AP score of 2x. So I think the reproduce issue on R-101 models is solved !

**Comment by ppwwyyxx** [comment]
Good progress! However it sounds like you're comparing to non-e2e versions in detectron? The e2e R-101-FPN mAP in detectron are 39.4/39.8 for 1x/2x.

**Comment by roytseng-tw** [comment]
@ppwwyyxx Oh, I did refer to the wrong table. Thanks for correcting me! Luckily, 39.2 seems close enough to 39.4, so I consider this issue is still solved. 🎉

**Comment by jason718** [comment]
@ppwwyyxx I actually reproduced and got a little big higher results than detectron.caffe2

Using  e2e R-101-FPN-1x:
![image](https://user-images.githubusercontent.com/3235510/43498202-b9bbc820-94fa-11e8-90fa-25f9b292e0df.png)

@roytseng-tw Awesome job!

**Comment by ppwwyyxx** [comment]
@jason718 it sounds like you're also comparing to a wrong number. Your results contain masks, so you want to compare with the maskrcnn versions in detectron, whose mAP are 40.0/40.9 for 1x/2x.
